internal/cli/task_cancel.go

```diff
@@ -0,0 +1,103 @@
+package cli
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type TaskCancelCommand struct {
+	*baseCommand
+
+	flagRunJobId string
+}
+
+func (c *TaskCancelCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+	); err != nil {
+		return 1
+	}
+	ctx := c.Ctx
+
+	var taskId string
+	if len(c.args) == 0 && c.flagRunJobId == "" {
+		c.ui.Output("Task ID or Run Job Id required.\n\n%s", c.Help(), terminal.WithErrorStyle())
+		return 1
+	} else {
+		taskId = c.args[0]
+	}
+
+	if c.flagRunJobId != "" && taskId != "" {
+		c.ui.Output("Both Run Job Id and Task Id was supplied, will look up by Task Id", terminal.WithWarningStyle())
+	}
+
+	var (
+		taskReq *pb.CancelTaskRequest
+	)
+
+	if taskId != "" {
+		taskReq = &pb.CancelTaskRequest{
+			Ref: &pb.Ref_Task{
+				Ref: &pb.Ref_Task_Id{
+					Id: taskId,
+				},
+			},
+		}
+	} else if c.flagRunJobId != "" {
+		taskReq = &pb.CancelTaskRequest{
+			Ref: &pb.Ref_Task{
+				Ref: &pb.Ref_Task_JobId{
+					JobId: c.flagRunJobId,
+				},
+			},
+		}
+	}
```

Comment:
```suggestion
taskReq = &pb.CancelTaskRequest{
  Ref: &pb.Ref_Task{},
}

if taskId != "" {
  taskReq.Ref.Ref = &pb.Ref_Task_Id{
    Id: taskId,
  }
} else if c.flagRunJobId != "" {
  taskReq.Ref.Ref = &pb.Ref_Task_JobId{
    JobId: c.flagRunJobId,
  }
}
```

I feel like refactoring this to only show what is different between the two branches improves readability. Feel free to discard

Reply:
Makes sense - the protobuf nesting gets a little out of hand 😄 I like this though.

Reply:
For sure the nesting hurts the eyes after a bit trying to identify that the difference here was `Ref_Task_Id` vs. `Ref_Task_JobId` :)